lib/instrumentation/generic-span.js

```diff
@@ -8,18 +8,37 @@ const TraceParent = require('traceparent')
 
 module.exports = GenericSpan
 
-function GenericSpan (agent, type, opts) {
+function GenericSpan (agent, ...args) {
+  let opts = {}
```

Comment:
I'd probably rewrite this to a ternary (and also guard against the `null` edge-case):

```js
const opts = typeof args[args.length - 1] === 'object'
  ? (args.pop() || {})
  : {}
```

Reply:
fixed